src/node_url.cc

```diff
@@ -407,29 +407,62 @@ int64_t ParseIPv4Number(const char* start, const char* end) {
   return strtoll(start, nullptr, R);
 }
 
+// https://url.spec.whatwg.org/#ipv4-number-parser
+bool IsIPv4NumberValid(std::string_view input) {
+  if (input.empty()) {
+    return false;
+  }
+
+  if (input.size() >= 2 && input[0] == '0') {
+    if (input[1] == 'X' || input[1] == 'x') {
+      if (input.size() == 2) {
+        return true;
+      }
+
+      return input.find_first_not_of("0123456789abcdefABCDEF", 2) ==
+             std::string_view::npos;
+    } else {
+      return input.find_first_not_of("01234567", 1) == std::string_view::npos;
+    }
+  }
+
+  return std::all_of(input.begin(), input.end(), ::isdigit);
+}
+
 // https://url.spec.whatwg.org/#ends-in-a-number-checker
-bool EndsInANumber(const std::string& input) {
-  std::vector<std::string> parts = SplitString(input, '.', false);
+bool EndsInANumber(const std::string_view input) {
```

Comment:
can you inline this?

Reply:
I guess we can. It's short.. doesn't have loops,..... looks like something alignable. I'll do it!

Reply:
done!